### ISSUE TITLE
PCT documentation documentation overview

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/pct#514]
**Overview of the different documentation tasks**

There are a number of documentation tasks

[ClassDocumentation](https://wiki.rssw.eu/pct/ClassDocumentation.md)
[HtmlDocumentation](https://wiki.rssw.eu/pct/HtmlDocumentation.md)
[XMLDocumentation](https://wiki.rssw.eu/pct/XMLDocumentation.md)
[JsonDocumentation](https://wiki.rssw.eu/pct/JsonDocumentation.md)
[ABLDuck](https://wiki.rssw.eu/pct/ABLDuck.md)

**Is it possible to create an overview of how the documentation tasks relate to each other**

* Which of these tasks are current and which, if any, are legacy.
* Is jsondocumentation, AFAIK the most recent in the set, the way to go or is it a specialized task for VS Code only
* What can be done with the jsondocumentation output, can it be used to generate a documentation site ?

I'm currently using ABL duck, I would love to use vscode as ABL editor, Should we switch to jsondocumentation or should we generate multiple documentation sets ?

**Comment by gquerret** [comment]
Documentation will be updated, but here is a short recap.
* ClassDocumentation is the initial task that converts source code into a structured document. It is based on ast.jar which is a problem for me (dependency problems, no documentation, no control over the API, ...). HtmlDoc and XmlDoc just read the output of this task, and generate HTML or XML.
* JsonDocumentation is the new task which depends on proparse. It is only used in VSCode for now. HtmlDoc and XmlDoc should be changed to read this task's output.
* AblDuck doesn't read the ClassDocumentation output, but directly connects to the ast.jar structure. I think that @spazzymoto already started rewriting the task to use JsonDoc, but it's not completed.

In short, JsonDocumentation is the future. All contributions are welcome !

**Comment by cverbiest** [comment]
Thank you, for the clarification.
I'll stick with AblDuck for our current documentation, and I'll add JsonDocumentation for vscode and future documentation.
